### DISCUSSION TITLE
Fixing broken dependency

### DISCUSSION THREAD

**Discussion by JorenB** [spack/spack#34327] [Q&A]
The default version of the package 'ncurses' (version 6.2 with abi=none) installed through spack causes several programs to crash with a segmentation fault. This issue does not occur for ncurses@6.2 with abi=5, so I want to replace the former package by the latter. However, many packages transitively depend on the faulty version of ncurses so I cannot simply uninstall it without simultaneously uninstalling all these dependents. Is there a way to fix this 'in place' so that the abi=none version gets replaced by the abi=5 and that all dependents subsequently point to the abi=5 version?

In principle, I could reinstall all those packages with the new abi=5 package as a dependency, but I would guess that a more straightforward solution exists.

**Comment by tgamblin** [comment]
If you know what you're doing, you can use `spack deprecate` for this.  It will symlink the old package directory to the new one so the old dependencies find the new thing.

```console
> spack install ncurses@6.2 abi=5
> spack deprecate ncurses@6.2 abi=none ncurses@6.2 abi=5   # symlinks first installation to point at second
```
